ceph_iscsi_config/lun.py

```diff
@@ -736,6 +738,113 @@ def remove_dev_from_lio(self):
 
                 break
 
+    @staticmethod
+    def valid_disk(logger, **kwargs):
+        """
+        determine whether the given image info is valid for a disk operation
+
+        :param logger: logger object
+        :param kwargs: See mode_vars below
+        :return: (str) either 'ok' or an error description
+        """
+
+        mode_vars = {"create"      : ['pool', 'image', 'size', 'count'],
+                     "resize"      : ['pool', 'image', 'size'],
+                     "reconfigure" : ['pool', 'image', 'max_data_area_mb'],
+                     "delete"      : ['pool', 'image']}
+
+        config_obj = Config(logger)
+        if config_obj.error:
+            return "Ceph iSCSI config object not available"
+        config = config_obj.config
```

Comment:
Yes, it works. The LUN() call which is done a little after the valid_disk call in the ceph-ansible patch:

https://github.com/mikechristie/ceph-ansible/commit/b9a4db8b4a18a543c0645821294bfaf54024e1a0

does a Config() too.

Let me actually repush this patch, because for the rbd-target-api valid_disk calls, the caller has a fresh config object. Also when we do LUN() calls we have a config object already too so there is not need to call Config() so many times in a row for this type of case.

Reply:
Note: I did an explicit test with a fresh config and it does work as expected. The new ceph settings like non standard cluster names will get used ok.